<a href="https://colab.research.google.com/github/PaulLipska/Codeing_Dojo_Project_2/blob/main/Project2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project2 - Part 1
Independent Project Investigation

##Healthcare Dataset Stroke Data

### Source of data:


https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset

### Description:


According to the World Health Organization (WHO) stroke is the 2nd leading cause of death globally, responsible for approximately 11% of total deaths.
This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status. Each row in the data provides relavant information about the patient.

### Target

Target is "Stroke" as function of contributing factors that share a commonality with people who have strokes

### What does one row represent?

The row represents a patient

In [11]:
import pandas as pd
url = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vSL6N5Ag95sphKFojQBp-pJBbmZeIFFMPizEDHCdXomjDZWJJlEy0vlaEFQJYCzYK1JytY6JfxdppHO/pub?output=csv'
df = pd.read_csv(url)
df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

### Is this a classification or regression problem?

This is a classification problem

###  How many features does the data have?

In [ ]:
df.shape

This dataset has 12 features

### How many rows are in this dataset?


5110 rows

### What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

In [ ]:
#Investigating Missing Values
df[df.isna().any(axis=1)].shape

In [ ]:
df.isna().sum()

Moderate risk in resolving missing values on what is a key indicator for risk.  Error in inputation could ruin investigation.

##Heart Failure Prediction Dataset

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

### Source of data:

https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction

### Description:

Cardiovascular diseases (CVDs) are the number 1 cause of death globally, taking an estimated 17.9 million lives each year, which accounts for 31% of all deaths worldwide. Four out of 5CVD deaths are due to heart attacks and strokes, and one-third of these deaths occur prematurely in people under 70 years of age. Heart failure is a common event caused by CVDs and this dataset contains 11 features that can be used to predict a possible heart disease.

People with cardiovascular disease or who are at high cardiovascular risk (due to the presence of one or more risk factors such as hypertension, diabetes, hyperlipidaemia or already established disease) need early detection and management wherein a machine learning model can be of great help.

### Target

The target for this investigation is heart disease as a function of predicting heart failure

### What does one row represent?

In [ ]:
import pandas as pd
filename = '/content/drive/MyDrive/CodingDojo/Beginning Data Science/Projects/heart.csv'
df = pd.read_csv(filename)
df.head(1)

The row is an investigation of individuals presenting an array of condtions that fall in the domanin of heart disease

### Is this a classification or regression problem?

This is a classification problem since it is investigating those with and without heart disease

### How many features does the data have?

In [ ]:
df.shape

There are 12 features

### How many rows are in this dataset?

There are 918 rows

### What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

In [ ]:
#Investigating Missing Values
df[df.isna().any(axis=1)].shape

In [ ]:
df.isna().sum()

In [ ]:
df.duplicated().sum()

There doesnt appear to be major issue with this data frame



---



#Project2 - Part 2: Heart Failure Prediction Data Set

### Import Library


In [ ]:
# import libraries
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

### Import Data Set

In [ ]:
import pandas as pd
filename = '/content/drive/MyDrive/CodingDojo/Beginning Data Science/Projects/heart.csv'
df = pd.read_csv(filename)
df.head()

In [ ]:
# Display the number of rows and columns for the dataframe
df.shape
print(f'There are {df.shape[0]} rows, and {df.shape[1]} columns.')
print(f'The rows represent {df.shape[0]} observations, and the columns represent {df.shape[1]-1} features and 1 target variable.')

In [ ]:
# Display the column names and datatypes for each column
# Columns with mixed datatypes are identified as an object datatype
df.dtypes

In [ ]:
# Display the column names, count of non-null values, and their datatypes
df.info()

In [ ]:
# Display the descriptive statistics for the numeric columns
df.describe(include="number") # or 'object'

### Clean the Data

#### Remove Unnecessary Columns


*   All Columns are needed




#### Remove Unnecessary Rows
Duplicates

In [ ]:
# Display the number of duplicate rows in the dataset
print(f'There are {df.duplicated().sum()} duplicate rows.')

####Inspect Column Datatypes for Errors

#Data Dictionary

#Attribute Information


1.   Age: age of the patient [years]
2.   Sex: sex of the patient [M: Male, F: Female]
3.   ChestPainType: chest pain type [TA: Typical Angina, ATA: Atypical Angina, NAP: Non-Anginal Pain, ASY: Asymptomatic]
4.   RestingBP: resting blood pressure [mm Hg]
5.   Cholesterol: serum cholesterol [mm/dl]
6.   FastingBS: fasting blood sugar [1: if FastingBS > 120 mg/dl, 0: otherwise]
7.   RestingECG: resting electrocardiogram results [Normal: Normal, ST: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV), LVH: showing probable or definite left ventricular hypertrophy by Estes' criteria]
8.   MaxHR: maximum heart rate achieved [Numeric value between 60 and 202]
9.   ExerciseAngina: exercise-induced angina [Y: Yes, N: No]
10.  Oldpeak: oldpeak = ST [Numeric value measured in depression]
11.  ST_Slope: the slope of the peak exercise ST segment [Up: upsloping, Flat: flat, Down: downsloping]
12.  HeartDisease: output class [1: heart disease, 0: Normal]














In [ ]:
df.dtypes

*Since there is not strict data dictionary on Kaggle this Attribute Information is the best possible guide.  All values seem to fit nicely within the given Dtype*

#### Inspect Column Names for Errors

In [ ]:
# Display column names
df.columns

No spaces or misspelled columns

####Examining Object Rows for Outliers

In [ ]:
sel_obj = list(df.select_dtypes(include='object'))
sel_obj

In [ ]:
df['Sex'].unique()

In [ ]:
df['ChestPainType'].unique()

In [ ]:
df['RestingECG'].unique()

In [ ]:
df['ExerciseAngina'].unique()

In [ ]:
df['ST_Slope'].unique()



*   No unusual string values found




#### Missing Values

In [ ]:
# Display the total number of missing values
print(f'There are {df.isna().sum().sum()} missing values.')



*   No missing values means no imputing will be needed for this investigation



####Create a Copy of the Dataset

In [ ]:
# Create a copy of the dataset for Modeling
df_ml = df.copy()

#Graphing Vizualizations

In [ ]:
sns.set_theme(style="ticks")

df = df_ml
sns.pairplot(df, hue="HeartDisease")
print('Overview for the dataframe in reference to the target column of HeartDisease')

In [ ]:
sns.boxplot(data=df, y='Cholesterol', x='Sex', notch=True)
plt.show()
print('You can see cholesterol is higher in Women but Men tend to defy categorical measurement due to higher number of outliers.')

# Project2 - Part 3: Correlation

In [ ]:
df_ml.head()

In [ ]:
df_ml.corr()

In [ ]:
sns.heatmap(df_ml.corr());

In [ ]:
f, ax = plt.subplots(figsize=(10,10)) # set the size that you'd like (width, height)
plt.bar(df_ml['Age'],df_ml['Oldpeak'])
plt.xlabel('Age Group')
plt.ylabel("Oldpeak")
plt.title('Level of Oldpeak per Age Group')
plt.show()

*oldpeak: ST depression induced by exercise relative to rest ('ST' relates to positions on the ECG plot. This is a numerical value as related to the above

### OldPeak has high correlation with Heart Disease the above graph illustrates the values among age group.  This diagnostic measures the slope positive or negative from an ekg after exercise

In [ ]:
sns.countplot(x="Oldpeak", hue="HeartDisease", data=df_ml)
sns.set(rc={'figure.figsize':(30,10)})

#sns.countplot(y='HeartDisease',hue='Oldpeak',data=df)
#sns.set(rc={'figure.figsize':(10,10)})
#plt.xticks(rotation=90)
#plt.title('Medals by Gender in each Discipline')
#plt.legend(loc=1)  # 1 is code for 'upper right'3


###Per the previous graph you can see incidence of heart disease as it correlates with different values of Oldpeak.  The value of 0 has the highest incidence of heart disease but also the highest incidence of testing negative.  As oldpeak rises we see those testing negative plummeting.  We can reasonably infer this is an inflection point between those who have heart disease and those who are about to get it.